packages/engine-core/src/library/types/Library.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  GetConfigOptions,
+  ConfigMetaFormat,
+  QueryEngineConfig,
+} from '../../common/types/QueryEngine'
+
+export type ConnectArgs = {
+  enableRawQueries: boolean
+}
+
+export type QueryEngineInstance = {
+  connect(connectArgs: ConnectArgs): Promise<void>
+  disconnect(): Promise<void>
+  /**
+   *
+   * @param request JSON.stringified `QueryEngineRequest | QueryEngineBatchRequest`
+   *
+   * @param headers JSON.stringified `QueryEngineRequestHeaders`
+   */
+  query(
+    request: string,
+    headers: string,
+    transactionId?: string,
+  ): Promise<string>
+  sdlSchema(): Promise<string>
```

Comment:
@williamluke4 

Any idea what is sdlSchema?

It’s not documented nor used I think.

Reply:
Can't remember, I'll have to ask Julius

Reply:
We can take care of this this in a separate PR